rust/operator-binary/src/discovery.rs

```diff
@@ -0,0 +1,160 @@
+use std::{collections::BTreeSet, num::TryFromIntError};
+
+use snafu::{OptionExt, ResultExt, Snafu};
+use stackable_hive_crd::{HiveCluster, HiveRole, APP_NAME, APP_PORT};
+use stackable_operator::{
+    builder::{ConfigMapBuilder, ObjectMetaBuilder},
+    k8s_openapi::api::core::v1::{ConfigMap, Endpoints, Service},
+    kube::{runtime::reflector::ObjectRef, Resource, ResourceExt},
+};
+
+use crate::controller::hive_version;
+
+#[derive(Snafu, Debug)]
+pub enum Error {
+    #[snafu(display("object {} is missing metadata to build owner reference", hive))]
+    ObjectMissingMetadataForOwnerRef {
```

Comment:
[rant]This the main reason I don't like snafu: this error is defined twice with different properties.[/rant]